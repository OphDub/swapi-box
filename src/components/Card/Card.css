.card {
  background: rgba(0, 0, 0, 0.8);
  border: solid 1px #FFFFFF;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 300px;
  min-height: 375px;
  max-height: 375px;
  margin: 10px 0; }

.card-info {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 50%;
  padding: 15px; }

.card-li {
  font-weight: 700; }

.favorited {
  background-color: rgba(65, 185, 225, 0.5); }

.favorites,
.people,
.planets,
.vehicles {
  padding: 15px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%; }

.favorites {
  background-color: rgba(153, 99, 153, 0.5); }

.people {
  background: rgba(162, 109, 162, 0.5); }

.planets {
  background: rgba(5, 74, 179, 0.5); }

.vehicles {
  background: rgba(0, 92, 18, 0.5); }
