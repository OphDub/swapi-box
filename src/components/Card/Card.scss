.card {
  border: yellow 1px solid;
}