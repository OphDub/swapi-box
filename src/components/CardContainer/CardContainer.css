.loading-vehicles {
  background-image: url("https://i.imgur.com/JSRLHzR.gif");
  background-repeat: no-repeat;
  width: 500px;
  height: 250px; }

.loading-planets {
  background-image: url("http://www.lasertimepodcast.com/wp-content/uploads/2015/03/anime-star-wars-laser-time.gif");
  background-repeat: no-repeat;
  width: 720px;
  height: 405px; }
