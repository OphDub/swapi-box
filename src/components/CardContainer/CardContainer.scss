.loading-vehicles {
  background-image: url("https://i.imgur.com/JSRLHzR.gif");
  background-repeat: no-repeat;
  width: 500px;
  height: 250px;
  display: flex;
  align-content: center;
}

.loading-planets {
  background-image: url("http://www.lasertimepodcast.com/wp-content/uploads/2015/03/anime-star-wars-laser-time.gif");
  background-repeat: no-repeat;
  width: 720px;
  height: 405px;
  display: flex;
  align-content: center;
}

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

