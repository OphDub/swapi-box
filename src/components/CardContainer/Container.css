.card-container {
  border: magenta 1px solid; }
