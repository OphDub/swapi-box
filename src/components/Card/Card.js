import React from 'react';
import './Card.css';
import { object, func } from 'prop-types';

const Card = ({ type, element, saveFavorite }) => {
  const keys = Object.keys(element).filter((key)=>
    key !== 'name' && key !== 'favorited'
  );
  const cardClass = element.favorited ? 'favorited' : ''

  const list = keys.map((key, index) => {
    let info = element[key];
    if (typeof element[key] === 'object') {
      info = element[key].join(', ')
    }

    return (
      <li key={`${Date.now()}${index}`}>
        {key}: {info}
      </li>
    );
  });

  return (
    <article className="card">
      <div className={type}>
        <h3>{element.name}</h3>
        <button onClick={() => saveFavorite(element)}
                className={cardClass}>
          Favorite
        </button>
      </div>
      <ul>
        {list}
      </ul>
    </article>
  );
};

Card.propTypes = {
  element: object.isRequired,
  saveFavorite: func.isRequired,
};

export default Card;