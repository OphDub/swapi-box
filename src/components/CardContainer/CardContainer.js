import React from 'react';
import './CardContainer.css';
import Card from '../Card/Card';
import { string, object, func, arrayOf } from 'prop-types';

const CardContainer = ({ type, data, saveFavorite, favorites }) => {
  let renderedCards;

  if (data.length === 0) {
    renderedCards = (<h3 className={type}> No {type} to show </h3>);
  } else {
    renderedCards = data.map((element, index) => {
      return (
        <Card element={element}
          type={type}
          key={`${Date.now()}${index}`}
          saveFavorite={saveFavorite}/>
      );
    });
  }

  return (
    <section className="card-container">
      {renderedCards}
    </section>
  );
};

CardContainer.propTypes = {
  type: string.isRequired,
  data: arrayOf(object).isRequired,
  saveFavorite: func.isRequired,
};

export default CardContainer;