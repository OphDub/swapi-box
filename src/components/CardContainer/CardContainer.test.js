/* eslint-disable */
import React from 'react';
import CardContainer from './CardContainer';
import { shallow, mount } from 'enzyme';
import { mockCleanPlanetData } from '../mock-data';
import Card from '../Card/Card';

describe('CARD CONTAINER', () => {
  it('should match the snapshot', () => {
    const mockDataArray = mockCleanPlanetData;
    const wrapper = shallow(<CardContainer data={mockDataArray}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render different contents based on the length of data it is provided', () => {
    const mockDataArray = mockCleanPlanetData;
    const mockFn = jest.fn();
    const wrapper = mount(<CardContainer type="planets" data={mockDataArray} saveFavorite={mockFn}/>);

    expect(wrapper.find('article').length).toEqual(1);
  });

  it('should render nothing if it is given an empty array', () => {
    const emptyArray = [];
    const mockFn = jest.fn();
    const wrapper = mount(<CardContainer type="planets" data={emptyArray} saveFavorite={mockFn}/>);

    expect(wrapper.find('article').length).toEqual(0);
  });
});