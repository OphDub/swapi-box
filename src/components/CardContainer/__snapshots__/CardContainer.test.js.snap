// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD CONTAINER should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    data={
        Array [
            Object {
              "count": 61,
              "next": "https://swapi.co/api/planets/?page=2",
              "previous": null,
              "results": Array [
                Object {
                  "climate": "temperate",
                  "created": "2014-12-10T11:35:48.479000Z",
                  "diameter": "12500",
                  "edited": "2014-12-20T20:58:18.420000Z",
                  "films": Array [
                    "https://swapi.co/api/films/6/",
                    "https://swapi.co/api/films/1/",
                  ],
                  "gravity": "1 standard",
                  "name": "Alderaan",
                  "orbital_period": "364",
                  "population": "2000000000",
                  "residents": Array [
                    "https://swapi.co/api/people/5/",
                    "https://swapi.co/api/people/68/",
                    "https://swapi.co/api/people/81/",
                  ],
                  "rotation_period": "24",
                  "surface_water": "40",
                  "terrain": "grasslands, mountains",
                  "url": "https://swapi.co/api/planets/2/",
                },
              ],
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          element={
                    Object {
                              "count": 61,
                              "next": "https://swapi.co/api/planets/?page=2",
                              "previous": null,
                              "results": Array [
                                Object {
                                  "climate": "temperate",
                                  "created": "2014-12-10T11:35:48.479000Z",
                                  "diameter": "12500",
                                  "edited": "2014-12-20T20:58:18.420000Z",
                                  "films": Array [
                                    "https://swapi.co/api/films/6/",
                                    "https://swapi.co/api/films/1/",
                                  ],
                                  "gravity": "1 standard",
                                  "name": "Alderaan",
                                  "orbital_period": "364",
                                  "population": "2000000000",
                                  "residents": Array [
                                    "https://swapi.co/api/people/5/",
                                    "https://swapi.co/api/people/68/",
                                    "https://swapi.co/api/people/81/",
                                  ],
                                  "rotation_period": "24",
                                  "surface_water": "40",
                                  "terrain": "grasslands, mountains",
                                  "url": "https://swapi.co/api/planets/2/",
                                },
                              ],
                            }
          }
          saveFavorite={undefined}
          type={undefined}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "15178060350330",
        "nodeType": "function",
        "props": Object {
          "element": Object {
            "count": 61,
            "next": "https://swapi.co/api/planets/?page=2",
            "previous": null,
            "results": Array [
              Object {
                "climate": "temperate",
                "created": "2014-12-10T11:35:48.479000Z",
                "diameter": "12500",
                "edited": "2014-12-20T20:58:18.420000Z",
                "films": Array [
                  "https://swapi.co/api/films/6/",
                  "https://swapi.co/api/films/1/",
                ],
                "gravity": "1 standard",
                "name": "Alderaan",
                "orbital_period": "364",
                "population": "2000000000",
                "residents": Array [
                  "https://swapi.co/api/people/5/",
                  "https://swapi.co/api/people/68/",
                  "https://swapi.co/api/people/81/",
                ],
                "rotation_period": "24",
                "surface_water": "40",
                "terrain": "grasslands, mountains",
                "url": "https://swapi.co/api/planets/2/",
              },
            ],
          },
          "saveFavorite": undefined,
          "type": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            element={
                        Object {
                                    "count": 61,
                                    "next": "https://swapi.co/api/planets/?page=2",
                                    "previous": null,
                                    "results": Array [
                                      Object {
                                        "climate": "temperate",
                                        "created": "2014-12-10T11:35:48.479000Z",
                                        "diameter": "12500",
                                        "edited": "2014-12-20T20:58:18.420000Z",
                                        "films": Array [
                                          "https://swapi.co/api/films/6/",
                                          "https://swapi.co/api/films/1/",
                                        ],
                                        "gravity": "1 standard",
                                        "name": "Alderaan",
                                        "orbital_period": "364",
                                        "population": "2000000000",
                                        "residents": Array [
                                          "https://swapi.co/api/people/5/",
                                          "https://swapi.co/api/people/68/",
                                          "https://swapi.co/api/people/81/",
                                        ],
                                        "rotation_period": "24",
                                        "surface_water": "40",
                                        "terrain": "grasslands, mountains",
                                        "url": "https://swapi.co/api/planets/2/",
                                      },
                                    ],
                                  }
            }
            saveFavorite={undefined}
            type={undefined}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "15178060350330",
          "nodeType": "function",
          "props": Object {
            "element": Object {
              "count": 61,
              "next": "https://swapi.co/api/planets/?page=2",
              "previous": null,
              "results": Array [
                Object {
                  "climate": "temperate",
                  "created": "2014-12-10T11:35:48.479000Z",
                  "diameter": "12500",
                  "edited": "2014-12-20T20:58:18.420000Z",
                  "films": Array [
                    "https://swapi.co/api/films/6/",
                    "https://swapi.co/api/films/1/",
                  ],
                  "gravity": "1 standard",
                  "name": "Alderaan",
                  "orbital_period": "364",
                  "population": "2000000000",
                  "residents": Array [
                    "https://swapi.co/api/people/5/",
                    "https://swapi.co/api/people/68/",
                    "https://swapi.co/api/people/81/",
                  ],
                  "rotation_period": "24",
                  "surface_water": "40",
                  "terrain": "grasslands, mountains",
                  "url": "https://swapi.co/api/planets/2/",
                },
              ],
            },
            "saveFavorite": undefined,
            "type": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
